app/controllers/FileUploadController.scala

```diff
@@ -94,6 +73,80 @@ class FileUploadController @Inject()(fileUploadConnector: FileUploadConnector,
       }
   }
 
+  def post: Action[AnyContent] = Action.async { implicit request =>
+    isAuthorised.flatMap {
+      case Right(userId) =>
+        getFile(request) match {
+          case file if (file.filename.length == 0) =>
+            Logger.info(s"[FileUploadController][post] No file selected")
+            Future.successful(BadRequest(views.html.file_upload(form, Messages("error.select.csv"))))
```

Comment:
I would return only the error message key here and convert it to an actual message on the view as a good general practice. (This applies to the other error cases in this pattern match)